modules/icgc-argo-workflows/payload/germlinevariant/resources/usr/bin/main.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+  Copyright (C) 2021,  Ontario Institute for Cancer Research
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Authors:
+    Edmund Su
+    Linda Xiang
+"""
+
+import os
+import sys
+import argparse
+import subprocess
+import json
+import re
+import hashlib
+import uuid
+import tarfile
+from datetime import date
+import copy
+from glob import glob
+import yaml
+import csv
+import io
+import shutil
+
+def calculate_size(file_path):
+    return os.stat(file_path).st_size
+
+
+def calculate_md5(file_path):
+    md5 = hashlib.md5()
+    with open(file_path, 'rb') as f:
+        for chunk in iter(lambda: f.read(1024 * 1024), b''):
+            md5.update(chunk)
+    return md5.hexdigest()
+
+def get_files_info(file_to_upload, date_str, analysis_dict, process_indicator,tool,new_dir,pipeline_info,tarball,data_type):
+    file_info = {
+        'fileSize': calculate_size(file_to_upload),
+        'fileMd5sum': calculate_md5(file_to_upload),
+        'fileAccess': 'controlled',
+        'info': {
+            'data_category': "Simple Nucleotide Variation",
+        }
+    }
+ 
+    if tarball=="false":
+        if tool=="deepvariant":
+            if re.match(r'.*.vcf.gz$', file_to_upload):
+                file_type = 'VCF'
+                file_info.update({'dataType': 'Raw %s Calls' % data_type})
+                file_info['info'].update({'analysis_tools': [{key.split(":")[-1]:pipeline_info[key]} for key in pipeline_info.keys()]})
+            elif re.match(r'.*.vcf.gz.tbi$', file_to_upload):
+                file_type = 'TBI'
+                file_info.update({'dataType': 'VCF Index'})
+                file_info['info'].update({'analysis_tools': [{key.split(":")[-1]:pipeline_info[key]} for key in pipeline_info.keys()]})
```

Comment:
We can move this pipeline_info into `workflow` section, and add individual tool here.